kidviz/views.py

```diff
@@ -1013,3 +1013,9 @@ def get_queryset(self):
             .prefetch_related('students') \
             .filter(owner=self.request.user, constructs=None)
             
+
+class RemoveDraft(LoginRequiredMixin, View):
+    def post(self, request, pk):
+        Observation.objects.filter(pk=pk, is_draft=True).delete()
```

Comment:
One more thing: this code allows to remove observation that is not owned by the requested user.